force-app/main/default/classes/CON_DeleteContactOverride_TEST.cls

```diff
@@ -551,4 +640,82 @@ private class CON_DeleteContactOverride_TEST {
             .withDateEstablished()
             .withScheduleTypeMultiplyValue();
     }
+
+    private class CON_DeleteContactOverrideStub implements System.StubProvider {
+
+        public Type classType = CON_DeleteContactOverrideSelector.class;
+
+        private Id contactId;
+        private Boolean canDeleteContact;
+        private Id accountId;
+        private Boolean canDeleteAccount;
+        private Boolean isAccountWithSingleContact;
+
+        final public String CONTACT_NAME = 'Test Contact';
+        final public String ACCOUNT_NAME = 'Test Account';
+
+        CON_DeleteContactOverrideStub(Id contactId, Boolean canDeleteContact, Id accountId, Boolean canDeleteAccount,
+            Boolean isHouseholdWithSingleContact)
+        {
+            this.contactId = contactId;
+            this.canDeleteContact = canDeleteContact;
+            this.accountId = accountId;
+            this.canDeleteAccount = canDeleteAccount;
+            this.isAccountWithSingleContact = isHouseholdWithSingleContact;
+        }
+
+        private Contact queryContact() {
+            Contact contactDetails = (Contact) JSON.deserialize('{' +
```

Comment:
Is there a reason JSON.deserialize is used here as opposed to instantiating a new Contact instance and assigning the field values directly?

Reply:
You should be able to assign an Account SObject directly to that field on the Contact record like this:

`contactDetails.Account = new Account(Name = ACCOUNT_NAME, npe01__SYSTEMIsIndividual__c = true);`

Reply:
@kenlewis-sfdo @bdvorachek It turns out that the actual problem with abandoning the JSON.deserialize approach is Contact.Name.  That field is not writeable and I have to have it to mock the way the confirmation dialog will behave.  Consequently, I believe we need to leave this as JSON.